### PR TITLE
Use 'flag' package instead of manually parsing arguments

### PR DESCRIPTION
**Pull request by gavinwade12**
This PR addresses issue #5. The method used and its flaws are described in [this thread](https://groups.google.com/forum/#!topic/golang-nuts/bkqf4-RDj5Q). Namely, `-v`, `--v`, `-version`, and `--version` are all equivalent, and flags like `-vu` do not work the same as `-v -u`.

### CODE REVIEW

**Comment by mstruebing** [comment]
Hey very nice and quick PR!
Thank you :+1: 

Just two little things: When I didn't enter a page I get an error message, but I think it would be better to print a help message then.
WDYT?

And the other https://golang.org/pkg/log/#Fatal `Fatal is equivalent to Print() followed by a call to os.Exit(1).`

You inserted 

```
log.Fatal("MESSAGE")
os.Exit(1)
```
inside your if statements two times. It would be nice to remove them.

Otherwise it's really fine, thank you.

**Comment by mstruebing** [comment]
Hey very nice and quick PR!
Thank you :+1:

Just two little things: When I didn't enter a page I get an error message, but I think it would be better to print a help message then.
WDYT?

And the other https://golang.org/pkg/log/#Fatal Fatal is equivalent to Print() followed by a call to os.Exit(1).

You inserted

log.Fatal("MESSAGE")
os.Exit(1)
inside your if statements two times. It would be nice to remove them.

Otherwise it's really fine, thank you.

File: main.go [hunk 1]
```diff
@@ -271,58 +268,47 @@ func printSinglePage(page string) {
...
-	case "-p":
-		if len(args) > 2 {
-			printPageForPlatform(args[1], args[2])
-		} else {
-			log.Fatal("ERROR: No platform provided or page provided")
+	} else if *platform != "" {
+		page := flag.Arg(0)
+		if page == "" {
+			log.Fatal("ERROR: no page provided")
+			os.Exit(1)
```
Comment (mstruebing) [inline main.go]: This line isn't necessary see: https://golang.org/pkg/log/#Fatal

File: main.go [hunk 2]
```diff
@@ -271,58 +268,47 @@ func printSinglePage(page string) {
...
-		if len(args) > 2 {
-			printPageForPlatform(args[1], args[2])
-		} else {
-			log.Fatal("ERROR: No platform provided or page provided")
+		printPageForPlatform(*platform, flag.Arg(0))
+	} else {
+		page := flag.Arg(0)
+		if page == "" {
+			log.Fatal("ERROR: no argument provided")
+			os.Exit(1)
```
Comment (mstruebing) [inline main.go]: This line isn't necessary see: https://golang.org/pkg/log/#Fatal

File: main.go [hunk 3]
```diff
@@ -271,58 +268,47 @@ func printSinglePage(page string) {
...
-	case "--platform":
-		if len(args) > 2 {
-			printPageForPlatform(args[1], args[2])
-		} else {
-			log.Fatal("ERROR: No platform provided or page provided")
+		printPageForPlatform(*platform, flag.Arg(0))
+	} else {
+		page := flag.Arg(0)
+		if page == "" {
+			log.Fatal("ERROR: no argument provided")
```
Comment (mstruebing) [inline main.go]: I'm not really sure but wouldn't it be nicer to print the help message in this case?

Comment (gavinwade12) [reply main.go]: In this case, after I print the help message, what do you think the exit status should be?

Comment (mstruebing) [reply main.go]: I would say 0, since no error is produced.

Comment (gavinwade12) [reply main.go]: That's what I was thinking as well. I just pushed up the changes addressing your review.

**Pull request merged**